ArjunaJTA/jta/classes/com/arjuna/ats/internal/jta/transaction/arjunacore/jca/TransactionImporterImple.java

```diff
@@ -167,11 +176,16 @@ public SubordinateTransaction getImportedTransaction(Xid xid)
 		if (xid == null)
 			throw new IllegalArgumentException();
 
-		SubordinateTransaction tx = _transactions.get(new SubordinateXidImple(xid));
+		TransactionImpleHolder holder = _transactions.get(new SubordinateXidImple(xid));
+
+		if (holder == null)
+			return null;
+
+		SubordinateTransaction tx = holder.getImported();
 
 		if (tx == null)
 			return null;
-		
+
```

Comment:
superfluous formatting change? but main point is can tx be null now? I would say not